### ISSUE TITLE
prefer-string-replace-all not failing when g flag is used

### ISSUE THREAD

**Issue by sinedied** [sindresorhus/eslint-plugin-unicorn#2194]
I have a few occurences in my code with a `replaceAll()` call and a regex with `g` flag and it's not reported as an issue.

`g` flags on a regex used with `replaceAll()` should be flagged as an error when the rule is enabled.

Few examples straight from my codebase where it's not reported/fixed:
```ts
filename.replaceAll(/[^\w-]/g, '_');

string_.replaceAll(/[A-Z]/g

s.replace('<', '&lt;').replace('>', '&gt;').replaceAll(/\r?\n/g, '<br>');

string_.replaceAll(/[A-Z]/g, (l) => `_${l}`).toUpperCase()

answer.replaceAll(/<<([^>]+)>>/g, (match, content) => { ... })
```

**Comment by fisker** [comment]
I don't understand what this mean. What's expected?

**Comment by fisker** [comment]
Hope this is not what you mean https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1948#issuecomment-1321246981

**Comment by sinedied** [comment]
Thanks @fisker for the link, it was an incorrect interpretation of the rule from my side so I'm closing this.
